cfme/tests/intelligence/reports/test_menus.py

```diff
@@ -116,13 +145,93 @@ def test_add_reports_to_available_reports_menu(appliance, request, group,
+):
+    """
+    https://bugzilla.redhat.com/show_bug.cgi?id=1535023
+    https://bugzilla.redhat.com/show_bug.cgi?id=1670293
+
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        caseimportance: medium
+        initialEstimate: 1/5h
+        startsin: 5.8
+        setup:
+            1. Create a new report called report01
+            2. Create a new user under EvmGroup-administrator called
+            testuser
+            3. "Edit Report Menus" and add the report01 under EvmGroup-
+            administrator's Tenant -> Tenant Quotas
+            4. Login using testuser and navigate to Reports
+        testSteps:
+            1. Check if the report01 is present under Tenant -> Tenant Quotas
+        expectedResults:
+            1. The report01 must be present under Tenant -> Tenant Quotas
+    """
+
+    folder, subfolder = "Tenants", "Tenant Quotas"
+    report_menus.move_reports(group, folder, subfolder, get_custom_report.menu_name)
+    with rbac_user:
+        rbac_report = appliance.collections.reports.instantiate(
+            type=folder, subtype=subfolder, menu_name=get_custom_report.menu_name
+        )
+        assert rbac_report.exists
+
+
+@pytest.mark.tier(1)
+@test_requirements.report
+@pytest.mark.meta(blockers=[BZ(1676638, forced_streams=["5.9", "5.10"])])
+@pytest.mark.parametrize("group", ["EvmGroup-super_administrator"])
+def test_reports_menu_with_duplicate_reports(
+    appliance, request, group, report_menus, custom_report_values
+):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Reporting
+        caseimportance: high
+        initialEstimate: 1/10h
+        startsin: 5.9
+        setup:
+            1.Create a custom report or copy an existing report.
+            2. Move the custom report to 'Tenants > Tenant Quotas' menu in user's current group
+        testSteps:
+            1. See if the report is available under 'Tenants > Tenant Quotas'.
+            2. Delete the custom report.
+            3. See if the report is available under 'Tenants > Tenant Quotas'.
+            4. Add a new report with the same menu_name.
+            5. See if the report is available under 'Tenants > Tenant Quotas'.
+        expectedResults:
+            1. Report must be visible.
+            2.
+            3. Report must not be visible.
+            4.
+            5. Report must not be visible.
+    """
+    custom_report_1 = appliance.collections.reports.create(**custom_report_values)
+    folder, subfolder = "Tenants", "Tenant Quotas"
+    report_menus.move_reports(group, folder, subfolder, custom_report_1.menu_name)
     report = appliance.collections.reports.instantiate(
-        type=folder, subtype=sub_folder, menu_name=custom_report.menu_name)
+        type=folder, subtype=subfolder, menu_name=custom_report_1.menu_name
+    )
     assert report.exists
+    custom_report_1.delete()
+    assert not report.exists
```

Comment:
@valaparthvi 
- I think we need teardown for ```custom_report_1``` because if test gets failed at ```assert report.exists```(line 232). Then it will not perform other operations. 
- But ```custom_report_1```  will remain in system. We can add like ```if custom_report_1.exists...```

Reply:
Teardown will not work here. Deleting the report and checking for its existence is a part of this test.